sdk/state/integration/helpers_test.go

```diff
@@ -93,63 +127,127 @@ func initAccounts(t *testing.T) (initiator Participant, responder Participant) {
 		txResp, err := client.SubmitFeeBumpTransaction(fbtx)
 		require.NoError(t, err)
 		responder.EscrowSequenceNumber = int64(txResp.Ledger) << 32
+
+		// add initial contribution
+		_, err = account.IncrementSequenceNumber()
+		require.NoError(t, err)
+
+		tx, err = txnbuild.NewTransaction(txnbuild.TransactionParams{
+			SourceAccount:        &account,
+			BaseFee:              txnbuild.MinBaseFee,
+			Timebounds:           txnbuild.NewTimeout(300),
+			IncrementSequenceNum: true,
+			Operations: []txnbuild.Operation{
+				&txnbuild.Payment{
+					Destination: responder.Escrow.Address(),
+					Amount:      stellarAmount.StringFromInt64(responder.Contribution),
+					Asset:       asset,
+				},
+			},
+		})
+
+		tx, err = tx.Sign(networkPassphrase, responder.KP)
+		require.NoError(t, err)
+		_, err = client.SubmitTransaction(tx)
+		require.NoError(t, err)
 	}
 	t.Log("Responder Escrow Sequence Number:", responder.EscrowSequenceNumber)
-	t.Log("Responder Contribution:", responder.Contribution)
+	t.Log("Responder Contribution:", responder.Contribution, "of asset:", asset.GetCode())
 	return initiator, responder
 }
 
-func initChannels(t *testing.T, initiator Participant, responder Participant) (initiatorChannel *state.Channel, responderChannel *state.Channel) {
+func initChannels(t *testing.T, client horizonclient.ClientInterface, initiator Participant, responder Participant) (initiatorChannel *state.Channel, responderChannel *state.Channel) {
 	// Channel constants.
 	const observationPeriodTime = 20 * time.Second
 	const averageLedgerDuration = 5 * time.Second
 	const observationPeriodLedgerGap = int64(observationPeriodTime / averageLedgerDuration)
 
+	initiatorEscrowAccount := state.EscrowAccount{
+		Address:        initiator.Escrow.FromAddress(),
+		SequenceNumber: initiator.EscrowSequenceNumber,
+	}
+	responderEscrowAccount := state.EscrowAccount{
+		Address:        responder.Escrow.FromAddress(),
+		SequenceNumber: responder.EscrowSequenceNumber,
+	}
+
 	initiatorChannel = state.NewChannel(state.Config{
```

Comment:
I think we can simplify this channel creation by combining with the initiatorEscrowAccount and responderEscrowAccount creation into a function called `makeChannel(localParticipant, remoteParticipant, isLocalInitiator)`. That would convert line 65-193 into two simple function calls -- and would reduce the copy pasted code.
Thoughts?

Reply:
PS: a function like `makeChannel` is something that could directly exist in the testing framework for use in tests (if we decide to build one)

Reply:
I created an `initEscrowAccount` method which removed the duplicated code. It should be easier to read now
https://github.com/stellar/experimental-payment-channels/pull/86/commits/d2c2c737ad46924bc6f929024fb143fa441ddf5a